src/lib.rs

```diff
@@ -99,9 +97,276 @@ use std::fmt;
 use std::iter::repeat;
 use std::result;
 
+/// A description of the options that a program can handle
+pub struct Options {
+    grps: Vec<OptGroup>
+}
+
+impl Options {
+    /// Create a blank set of options
+    pub fn new() -> Options {
+        Options {
+            grps: Vec::new()
+        }
+    }
+
+    /// Create a generic option group, stating all parameters explicitly
+    pub fn add_opt<'a>(&'a mut self, short_name: &str, long_name: &str, desc: &str,
```

Comment:
I'm happy to change that if it's the standard, but it seemed a bit ambiguous to me, since one is adding in multiple options (I would tend to read "opt(x)" as "set the only option to x" rather than "add an option x to the list of options).